api/internal/git/repospec.go

```diff
@@ -118,6 +119,20 @@ func parseGitURL(n string) (
 		path, gitRef, gitTimeout, gitSubmodules = peelQuery(n[index+len(gitDelimiter)+len(orgRepo):])
 		return
 	}
+	if strings.HasPrefix(n, fileScheme) {
```

Comment:
I did not populate `GitSuffix` because of this line, that drops the `GitSuffix` from the `Origin` https://github.com/kubernetes-sigs/kustomize/blob/14947e449bde0785cd6cc3ee90562095b34f1985/api/resource/origin.go#L55. Looking at it again I think it was an unintentional omission.

Reply:
Ah, now I remember there was another more problematic reason. `GitSuffix` **defaults** to `".git"`. It would be automatically included in the `CloneSpec()`. This would make the clone fail for `file://` URLs where `.git` is not in the directory name.